src/fileops/httpiochain.hpp

```diff
@@ -38,6 +38,14 @@ class ContentProvider;
     }while(0)
 
 
+#define CHAIN_FORWARD_VOID(X) \
+        do{ \
+        if(_next.get() != NULL){ \
+            _next->X; return; \
+        } \
+        throw DavixException(davix_scope_io_buff(), StatusCode::OperationNonSupported, "I/O operation not supported"); \
+    }while(0)
+
```

Comment:
Is this void return type really needed? Can't the other functions return the number of bytes they processed or 0 for success?

The `CHAIN_FORWARD` macro is not reader-friendly already. I don't want this mechanism to multiply based on return type

Reply:
Nothing is needed. I just a) didn't have a value to return, and b) he caller didn't need one. I suppose we can return a nonsense value the additon of a OID method is truly bothersome. Though going that way is just as icky.

Reply:
Can we go by that way? I agree both ways are unfulfilling, but I'd prefer only one `CHAIN_FORWARD` beast in the codebase. Thanks!

Reply:
Just pushed a patch to remove it.